src/Parser/DeathKnight/Blood/Modules/Features/UnendingThirstTracker.js

```diff
@@ -12,28 +12,35 @@ class UnendingThirstTracker extends Analyzer {
     combatants: Combatants,
   };
 
-  RecentDSCounter = 0;
-  DScost = 0;
+  bloodShieldRefreshed = 0;
+  totalDSCounter = 0;
 
   on_byPlayer_cast(event) {
     const spellId = event.ability.guid;
     if (spellId !== SPELLS.DEATH_STRIKE.id) {
       return;
     }
-    if (this.combatants.selected.hasBuff(SPELLS.BLOOD_SHIELD.id, event.timestamp)) {
-      this.RecentDSCounter += 1;
-      this.DScost = event.classResources[0].cost;
-      console.log('DScost is ', this.DScost);
+    this.totalDSCounter += 1;
+  }
+
+  //Death Strike causes the buff blood shield. If blood shield is refreshed it means death strike was casted when the blood shield buff was active.
+  on_byPlayer_refreshbuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.BLOOD_SHIELD.id) {
+      this.bloodShieldRefreshed += 1;
     }
   }
 
+
+
+
   statistic() {
     return (
       <StatisticBox
         icon={<SpellIcon id={SPELLS.UNENDING_THIRST.id} />}
-        value={this.RecentDSCounter}
-        label="Empowered Death Strike"
-        tooltip="If you're looking to increase your dps as the cost of defense try increasing this."
+        value={`${this.bloodShieldRefreshed} out of ${this.totalDSCounter}`}
```

Comment:
Consider either showing the percentage of empowered death strikes here, or showing the total additional damage (or dps) gained through unending thirst. If you do either, you can put the exact "X out of Y" in the tooltip.

Reply:
This isnt a bad idea. I will look into seeing if i can show the additional damage as a dps gain.